### PR TITLE
Enable students to drop their submissions when quest is hidden

### LINKED ISSUE #254: Student's can't drop submissions from quests that have become not visible
https://hackerspace.sd72.bc.ca/media/attachments/postman_message/316/Screenshot_from_2019-12-12_12-34-50.png

If a student has submitted a quest, and then that quest is made not visible, the submission stull appears for them, but clicking it leads to error page

### PR DESCRIPTION
**Pull request by yujinyuz**
Closes #254

### CODE REVIEW

File: src/quest_manager/templates/quest_manager/submission.html [hunk 1]
```diff
@@ -71,17 +71,19 @@ <h3 class="panel-title">Quest Submission Form</h3>
...
+              <span id="draft-spin" style="display:none"><i class="fa fa-spinner fa-spin fa-fw"></i><span class="sr-only">Saving draft...</span></span>
+              <span id="draft-check" style="display:none" class="text-success"><i class="fa fa-check fa-fw"></i><span class="sr-only">Draft saved successfully</span></span>
+            </a>
+            {% if submission.is_approved %}
+              <input type='submit' name='comment' class='btn btn-primary' value='Add Comment' />
+            {% else %}
+              <input id='tour-submit-quest' type='submit' name='complete' class='btn btn-primary' value='Submit Quest for Completion' />
+            {% endif %}
           {% endif %}
+          <a class="btn btn-danger" href="{% url 'quests:drop' submission.id %}" role="button">Drop Quest</a>
```
Comment (tylerecouture) [inline src/quest_manager/templates/quest_manager/submission.html:86]: Why is this moved?  Won't this show students the "drop quests" button for quests that have already been approved?  They shouldn't be able to drop approved quests.

If my test suite wasn't so bad it would have caught this! haha

**Comment by tylerecouture** [comment]
Everything else looks great!  Thanks for the new tests and the code comments, made it mush easier to understand what you were doing!

File: src/quest_manager/templates/quest_manager/submission.html (hunk 1 above)
Comment (yujinyuz) [reply src/quest_manager/templates/quest_manager/submission.html:86]: Yeah. I forgot about it. I'll hide this from them and also change the queryset as well
